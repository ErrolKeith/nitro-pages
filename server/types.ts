export interface BusinessClientGlobals {
  businessId: string; //uuidv4
  sites: string[]; //uuidv4[]
  companyName: string;
  phoneNumber: string;
}

export interface SiteGlobals {
  businessId: string; //uuidv4
  siteId: string;
  companyName: string;
  phoneNumber: string;
  blogSettings: {
    blogTitle: string;
  };
}
