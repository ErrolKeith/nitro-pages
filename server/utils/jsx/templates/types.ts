const templates = ["home", "page", "internal", "404", "index", "post"] as const;
type TemplateType = (typeof templates)[number];

export interface TemplateConfig {
  type: TemplateType;
  siteId: string;
  companyName: string;
  seoTitle: string;
  content: PageContentSection[];
}
