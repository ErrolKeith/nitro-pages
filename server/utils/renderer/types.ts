import { SiteGlobals } from "~/types";
import { TemplateType } from "../jsx/template";

export interface PageContext {
  template: TemplateType;
  slug: string;
  title: string;
  categories: string[];
  content: PageContentSection[];
}

export interface BaseRenderConfig {
  template: TemplateType;
  clientGlobals: SiteGlobals;
  pageContext: PageContext;
}

export interface PageRenderConfig extends BaseRenderConfig {
  template: "page" | "internal" | "post" | "index";
}

export type HomeRenderConfig = Omit<BaseRenderConfig, "template">;
