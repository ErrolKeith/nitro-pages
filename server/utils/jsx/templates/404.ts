import { TemplateConfig } from "./types";

export function make404Template(config: TemplateConfig) {
  const { siteId, seoTitle, content } = config;

  return makeHtml({
    props: {
      ["data-site-id"]: siteId,
    },
    children: [
      makeHead({ props: {}, children: [makeTitle({ title: seoTitle })] }),
      makeMain({ props: {}, children: [] }),
    ],
  });
}
