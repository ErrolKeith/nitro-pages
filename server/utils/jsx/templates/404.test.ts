import { it, expect } from "vitest";
import { make404Template } from "./404";
import { TemplateConfig } from "./types";

it("can make a 404 page", () => {
  const test404TemplateConfig: TemplateConfig = {
    type: "404",
    siteId: "124246796763",
    companyName: "sample company name",
    seoTitle: "sample 404",
    content: [],
  };
  const template404Test = make404Template(test404TemplateConfig);
  expect(template404Test).toMatchSnapshot();
});
