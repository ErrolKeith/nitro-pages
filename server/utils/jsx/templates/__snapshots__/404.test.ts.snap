// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make a 404 page 1`] = `
<html
  data-site-id="124246796763"
>
  <head>
    <title>
      sample 404
    </title>
  </head>
  <main />
</html>
`;
